### PR TITLE
Issue/2025 When enter/return key is pressed the input is saved

### PR DESCRIPTION
**Pull request by RachelDau**
Since onFormSubmit function was already implemented to listen and trigger when the enter key is pressed in practice mode, I went there and added code in the preexisting if statement that sends the response to be saved in assessment mode. Now a keyboard user is able to use the enter key to save numeric input responses.

### CODE REVIEW

**Review by FrenjaminBanklin — CHANGES_REQUESTED** [review]
There are some browser-based issues that are now evident as a result of this change.

The answer is now saved during actual assessments, so that's taken care of. But in Chrome, the input now loses focus, meaning users reliant on a screen reader are required to tab back into the input (and possibly navigate through other tabbables to get there).

In Firefox and Safari this isn't an issue - the input remains focused, but we'll have to do something to address Chrome.

You'll notice in the `componentDidUpdate` of the `Question` viewer component that a switch is checking the state of `this.nextFocus` and shifting focus around the page as necessary. I would suggest creating a new condition here to refocus the answer input upon pressing the 'Enter' key.

**Comment by RachelDau** [comment]
OnEnterPress function was added to delay the refocus of numeric input on enter press in order to to avoid refocusing when a lot of rendering is going on. Unit test added to test that focus is executed at the delay. Now upon enter press the input should retain focus and the answer be saved on all browser types.

File: packages/obonode/obojobo-chunks-question/viewer-component.js [hunk 1]
```diff
@@ -242,6 +248,12 @@ export default class Question extends React.Component {
 		this.setState({ isFlipping: true })
 		setTimeout(() => this.setState({ isFlipping: false }), DURATION_FLIP_TIME_MS)
 	}
+	//delays refocus on input to account for saved text rendering causing a visual loss of focus
+	onEnterPress() {
+		setTimeout(() => {
+			this.assessmentComponentRef.current.inputRef.current.focus()
+		}, 80)
```
Comment (FrenjaminBanklin) [inline packages/obonode/obojobo-chunks-question/viewer-component.js:255]: This is still kind of a magic number. I'd set it to a constant at the top of the file and use that here instead, like how `DURATION_FLIP_TIME_MS` is used above.

**Review by FrenjaminBanklin — CHANGES_REQUESTED** [review]
One really minor nitpick, see below.

Otherwise this looks like it's good to go.

File: packages/obonode/obojobo-chunks-question/viewer-component.js (hunk 1 above)
Comment (RachelDau) [reply packages/obonode/obojobo-chunks-question/viewer-component.js]: Const added. It is still kinda a magic number though. It was the lowest number that reliably delayed refocus enough to retain focus on all browsers.

**Review by FrenjaminBanklin — APPROVED** [review]

**Pull request merged**